src/Storages/StorageMaterializeMySQL.h

```diff
@@ -0,0 +1,32 @@
+#pragma once
+
+#include <Storages/IStorage.h>
+#include <Databases/MySQL/DatabaseMaterializeMySQL.h>
+
+namespace DB
+{
+
+class StorageMaterializeMySQL final : public ext::shared_ptr_helper<StorageMaterializeMySQL>, public IStorage
+{
+    friend struct ext::shared_ptr_helper<StorageMaterializeMySQL>;
+public:
+    String getName() const override { return "MaterializeMySQL"; }
+
+    bool supportsFinal() const override { return nested_storage->supportsFinal(); }
+    bool supportsSampling() const override { return nested_storage->supportsSampling(); }
+
```

Comment:
Maybe implement more methods, such as `optimize(...)`, `getActionLock(...)`, `supportsPrewhere()`, `mayBenefitFromIndexForIn(...)`, `checkData(...)`, `totalRows()`, etc? (can be done in another PR)

Reply:
Also it's useful to show information about internal MergeTree tables in `system.parts`

Reply:
I will implement them in another PR.